lib/trento/software_updates/discovery.ex

```diff
@@ -41,7 +41,16 @@ defmodule Trento.SoftwareUpdates.Discovery do
     do:
       {:ok,
        Hosts.get_all_hosts()
-       |> Enum.map(&discover_host_software_updates/1)
+       |> Enum.map(fn
+         %HostReadModel{id: host_id, fully_qualified_domain_name: fully_qualified_domain_name} ->
+           case discover_host_software_updates(host_id, fully_qualified_domain_name) do
+             {:error, error} ->
+               {:error, host_id, error}
+
+             _ = success ->
```

Comment:
I think that having here `:ok` pattern matching is more visual. What do you think?
```
{:ok, host_id, system_id, relevant_patches} ->
   {:ok, host_id, system_id, relevant_patches}

# or
{:ok, _, _, _} = success ->
  success
```

It is more obvious, and if for some reason we add any other `:error` return to the `discovery_host_software_updates`, we would still be safe